service/controller/resource/certificates/resource.go

```diff
@@ -66,17 +66,18 @@ func (sc *secretCopier) ToCR(v interface{}) (metav1.Object, error) {
 		return nil, microerror.Mask(err)
 	}
 
+	var sourceData map[string][]byte
 	sourceSecret, err := sc.getSource(context.TODO(), v)
-	if err != nil {
-		return nil, microerror.Mask(err)
+	if err == nil {
```

Comment:
The drawback of this is that in situation when a cluster is being **created** but the secret doesn't exist, we won't see an error here and we'll just create an empty secret which will lead to a failure further down the line.

Don't know what creates those source secret objects and how likely is it that they wouldn't be there when creating a cluster.

Looking forward to any comments on what y'all think about this.

Reply:
IIUC, those are created by the cert operator when the cluster operator creates cert config CRD. I would rather we handle the error when it happens and not create a dummy secret. Could we not wait till the secret is populated before creating the prometheus CR instead?

Reply:
That would be cool :)

Reply:
Can we move that outside the PR so we can approve it and fix the CRD issue?

Reply:
Yip, so https://github.com/giantswarm/prometheus-meta-operator/pull/115 is for dealing with the deletions and finalizer and refactoring, I'll update this to just deal with the `v1alpha3` resources.

Reply:
Done, reverted here.